torch/_dynamo/bytecode_analysis.py

```diff
@@ -24,9 +25,23 @@
 stack_effect = dis.stack_effect
 
 
+def get_indexof(insts):
+    """
+    Get a mapping from instruction memory address to index in instruction list.
+    Additionally checks that each instruction only appears once in the list.
+    """
+    indexof = {}
+    for i, inst in enumerate(insts):
+        if id(inst) in indexof:
+            breakpoint()
+        assert id(inst) not in indexof
+        indexof[id(inst)] = i
```

Comment:
Most places that this code replaces use `id(inst)` instead. My intuition is that we want to make sure each instruction object gets its own key in case the equality/hash functions change in the future. @jansel do you recall the original intent?